core/connection/manager.go

```diff
@@ -653,6 +653,8 @@ func (m *connectionManager) Disconnect() error {
 		return ErrNoConnection
 	}
 
+	m.eventBus.Unsubscribe(sleep.AppTopicSleepNotification, m.handleSleepEvent)
```

Comment:
Avoid subscribing on runtime. To avoid problems like this, or multiple subscriptions, races, memory leaks.